usaspending_api/search/filters/postgres/defc.py

```diff
@@ -1,17 +1,15 @@
 from django.db.models import Q
 
-from usaspending_api.common.elasticsearch.search_wrappers import AwardSearch
-from usaspending_api.common.query_with_filters import QueryWithFilters
+from usaspending_api.awards.models import FinancialAccountsByAwards
 
 
 class DefCodes:
     underscore_name = "def_codes"
 
     @classmethod
     def build_def_codes_filter(cls, queryset, filter_values):
-        query = {"def_codes": filter_values}
-        filter_query = QueryWithFilters.generate_awards_elasticsearch_query(query)
-        search = AwardSearch().filter(filter_query)
-        response = search.handle_execute()
-        award_ids = [res.to_dict()["award_id"] for res in response]
-        return Q(award_id__in=list(award_ids))
+        subquery = FinancialAccountsByAwards.objects.filter(disaster_emergency_fund__in=filter_values).values(
+            "award__id"
+        )
+        q = Q(award_id__in=subquery)
```

Comment:
This seems OK. 

I'm curious if it was possible to run performance tests with a sizable amount of data? Currently in QAT 983,990 Award IDs would be returned by the subquery. Number drops to 162,744 if only distinct are returned. I expect both of these numbers to be higher in production by next week.

Adding distinct could help or hurt. 
Could also attempt to use an OuterRef but that could open a can of worms.

Reply:
I tested against both DEV and QAT - DEV has 4020730 returned by the subquery, and the performance was ~5 times worse when using `.distinct()`.

Reply:
(non-distinct) | 1065 ms | 1019 ms | 1051 ms | 1506 ms
-- | -- | -- | -- | --
(distinct) | 5.44 s | 7.02 s | 4.49 s | 4.40 s